pennylane_lightning/core/src/simulators/lightning_tensor/tncuda/TNCudaBase.hpp

```diff
@@ -60,133 +57,194 @@ class TNCudaBase : public TensornetBase<PrecisionT, Derived> {
     using CFP_t = decltype(cuUtil::getCudaType(PrecisionT{}));
     using ComplexT = std::complex<PrecisionT>;
     using BaseType = TensornetBase<PrecisionT, Derived>;
-    SharedTNCudaHandle handle_;
+
+  protected:
+    // Note both maxBondDim_ and bondDims_ are used for both MPS and Exact
+    // Tensor Network. Per Exact Tensor Network, maxBondDim_ is 1 and bondDims_
+    // is {1}. Per Exact Tensor Network, setting bondDims_ allows call to
+    // appendInitialMPSState_() to append the initial state to the Exact Tensor
+    // Network state.
+    const std::size_t
+        maxBondDim_; // maxBondDim_ default is 1 for Exact Tensor Network
+    const std::vector<std::size_t>
+        bondDims_; // bondDims_ default is {1} for Exact Tensor Network
+
+  private:
+    const std::vector<std::vector<std::size_t>> sitesModes_;
+    const std::vector<std::vector<std::size_t>> sitesExtents_;
+    const std::vector<std::vector<int64_t>> sitesExtents_int64_;
+
     SharedCublasCaller cublascaller_;
-    cudaDataType_t typeData_;
-    DevTag<int> dev_tag_;
-    cutensornetComputeType_t typeCompute_;
-    cutensornetState_t quantumState_;
-    cutensornetStatePurity_t purity_ =
-        CUTENSORNET_STATE_PURITY_PURE; // Only supports pure tensor network
-                                       // states as v24.03
 
     std::shared_ptr<TNCudaGateCache<PrecisionT>> gate_cache_;
     std::set<int64_t> gate_ids_;
 
     std::vector<std::size_t> identiy_gate_ids_;
 
+    std::vector<TensorCuda<PrecisionT>> tensors_;
+    std::vector<TensorCuda<PrecisionT>> tensors_out_;
```

Comment:
what's tensors_out_?

Reply:
Tensors appended to the graph for the final layer of circuit of MPS calculations.